crates/forge_api/src/forge_api.rs

```diff
@@ -44,8 +45,8 @@ impl<F: Services + Infrastructure> API for ForgeAPI<F> {
         self.suggestion_service.suggestions().await
     }
 
-    async fn tools(&self) -> Vec<ToolDefinition> {
-        self.app.tool_service().list()
+    async fn tools(&self, mcp: HashMap<String, McpConfig>) -> anyhow::Result<Vec<ToolDefinition>> {
```

Comment:
Should take a conversation id.
```suggestion
    async fn tools(&self, conversation: ConversationId) -> anyhow::Result<Vec<ToolDefinition>> {
```